common/src/autogluon/common/utils/lite.py

```diff
@@ -0,0 +1,14 @@
+from .utils import get_autogluon_metadata
+
+
+def disable_if_lite_mode(ret=None):
+    def inner(func):
+        def do_nothing(*args, **kwargs):
+            if callable(ret):
+                return ret(*args, **kwargs)
+            return ret
+        metadata = get_autogluon_metadata()
+        if metadata['lite']:
+            return do_nothing
+        return func
+    return inner
```

Comment:
Have we checked if this has any meaningful overhead / slowdown to inference throughput when lite_mode is disabled due to having to repeatedly call `get_autogluon_metadata()`? I'd expect not, but @liangfu might be able to run a quick sanity check on mainline compared to this PR.

Reply:
Perhaps a better approach is to just have a dedicated `is_lite_mode_enabled()` function so it avoids having to generate all the extra information present in `metadata`.